src/jvm/main/org/jetbrains/kotlinx/lincheck/paramgen/ParameterGenerator.java

```diff
@@ -31,6 +31,9 @@
 public interface ParameterGenerator<T> {
     T generate();
 
+    default void restart() {
```

Comment:
Do you really need a default implementation?

Reply:
I this it's convenient as sometimes there is no sence in `resetRange` for some test generators.